### PR TITLE
Added Checkbox changed Primary button and TextField

### PR DESCRIPTION
**Pull request by Venkat-corebts**
> **Note**
>This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [ ] Update **documentation** as appropriate (e.g [README.md](../blob/main/README.md), and [**Architecture.md**](../blob/main/docs/Architecture.md), etc.)
- [x] **Run the app** and try the changes
- [x] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

> **Note**
> It is good practice to provide before and after UI  **screenshots** in the description of this PR. This is only applicable for changes that modify the UI.

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [x] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [ ] Review the **documentation**, [**README.md**](../blob/main/README.md), and [**Architecture.md**](../blob/main/docs/Architecture.md) as appropriate
- [x] **Run the app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Button.kt [hunk 1]
```diff
@@ -40,24 +50,30 @@ fun PrimaryButton(
...
+    buttonColor: Color = MaterialTheme.colorScheme.primary,
+    textColor: Color = MaterialTheme.colorScheme.onPrimary,
 ) {
     Button(
         shape = RectangleShape,
+        enabled= enabled,
+        modifier = modifier
+            .padding(outerPaddingValues)
+            .shadow(
+                Color.Black,
```
Comment (HonzaR) [inline ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Button.kt:65]: These color definitions could be part of the `Colors` file. We could then access them with `ZcashTheme.colors`.

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Button.kt [hunk 2]
```diff
@@ -40,24 +50,30 @@ fun PrimaryButton(
...
+                borderRadius = 0.dp,
+                offsetX = 20.dp,
+                offsetY = 20.dp,
+                spread = 10.dp,
+                blurRadius = 0.dp,
+                stroke = textColor != Color.Black,
+            )
+            .defaultMinSize(230.dp, 50.dp)
```
Comment (HonzaR) [inline ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Button.kt:73]: Should we move these sizes to the `Dimens` so we can customize them once we decide to support different screen sizes, orientations, or directions?

**Review by HonzaR — COMMENTED** [review]
Thank you for the proposed changes. Having the necessary UI components defined reusably in the `ui-design-lib` module is great! I left a few minor comments inline.

As this is our first PR, I note a few organizational things:
- In case you've missed it, we provide [contribution guidelines](https://github.com/zcash/secant-android-wallet/blob/main/docs/CONTRIBUTING.md), which are good to follow.
-  It's also good to prepare a PR against an existing issue (or create one with the task description) and connect it to the PR
- Don't hesitate to use the Author's checkboxes in a new PR
- For this type of PR, it'd also be helpful to see before/after screenshots of such UI changes, which is not necessary now, as I built and checked UI changes in the app.
- Finally, please provide your e-mail address so we can grant you access to some internal docs describing a few more things, ideally via your contact person with ECC. 

Thank you.

**Review by HonzaR — APPROVED** [review]

**Comment by Venkat-corebts** [comment]
Hi @HonzaR, when can I expect this branch to be merged as these are reusable components I need to raise PR for dependent issues

**Comment by HonzaR** [comment]
@Venkat-corebts Done. Note that the deployment to the Google Play internal testing will fail because of our latest deployment trouble described here https://github.com/zcash/secant-android-wallet/issues/935. But it shouldn't block you.

**Pull request merged**